library/items/item-group/src/main/java/org/quiltmc/qsl/itemgroup/impl/QuiltCreativeGuiComponents.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.itemgroup.impl;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+
+import net.minecraft.client.gui.screen.ingame.CreativeInventoryScreen;
+import net.minecraft.client.gui.widget.ButtonWidget;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.item.ItemGroup;
+import net.minecraft.text.LiteralText;
+import net.minecraft.text.Text;
+import net.minecraft.text.TranslatableText;
+import net.minecraft.util.Identifier;
+
+public class QuiltCreativeGuiComponents {
+	private static final Identifier BUTTON_TEX = new Identifier("qsl-items-item-group", "textures/gui/creative_buttons.png");
+	public static final Set<ItemGroup> COMMON_GROUPS = new HashSet<>();
+
+	static {
+		COMMON_GROUPS.add(ItemGroup.SEARCH);
+		COMMON_GROUPS.add(ItemGroup.INVENTORY);
+		COMMON_GROUPS.add(ItemGroup.HOTBAR);
+	}
+
+	public static class ItemGroupButtonWidget extends ButtonWidget {
+		CreativeGuiExtensions extensions;
+		CreativeInventoryScreen gui;
+		Type type;
+
+		public ItemGroupButtonWidget(int x, int y, Type type, CreativeGuiExtensions extensions) {
+			super(x, y, 11, 10, type.text, (bw) -> type.clickConsumer.accept(extensions));
+			this.extensions = extensions;
+			this.type = type;
+			this.gui = (CreativeInventoryScreen) extensions;
+		}
+
+		@Override
+		public void render(MatrixStack matrixStack, int mouseX, int mouseY, float float_1) {
```

Comment:
What happened to those params?
```suggestion
		public void render(MatrixStack matrices, int mouseX, int mouseY, float delta) {
```